actions/lib/action.py

```diff
@@ -22,18 +23,20 @@ def __init__(self, config):
             'aws_access_key_id': None,
             'aws_secret_access_key': None
         }
-        self.user_data_file = config.get('st2_user_data', None)
         self.debug = config.get('debug', False)
 
         self.userdata = None
-
-        # Read in default user data
-        if self.user_data_file:
-            try:
-                with open(self.user_data_file, 'r') as fp:
-                    self.userdata = fp.read()
-            except IOError as e:
-                self.logger.error(e)
+        if config.get('st2_user_data', None):
+            # Check if string "looks" like a normal absolute path
+            if os.path.isabs(config['st2_user_data']) and '\n' not in config['st2_user_data']:
```

Comment:
Linux path can have new line in it.

However need to distinguish somehow file path vs raw ec2 user data. Any better ideas?

Reply:
I cannot think of a better way to check. I expect occurrence of `\n` in a file path will be very rare (if ever).